drivers/display/display_nt35510.c

```diff
@@ -0,0 +1,417 @@
+	ret += nt35510_write_reg_val(dev, NT35510_CMD_WRDISBV, 0x7f);
+
+	/* defaut is 0, try 0x2C - Brightness Control Block, Display Dimming & BackLight on */
+	// ret += nt35510_write_reg(dev, NT35510_CMD_WRCTRLD, &nt35510_reg32[1], 1);
+	ret += nt35510_write_reg_val(dev, NT35510_CMD_WRCTRLD, 0x2c);
+
+	/* defaut is 0, try 0x02 - image Content based Adaptive Brightness [Still Picture] */
+	// ret += nt35510_write_reg(dev, NT35510_CMD_WRCABC, &nt35510_reg33[1], 1);
+	ret += nt35510_write_reg_val(dev, NT35510_CMD_WRCABC, 0x2);
+
+	/* defaut is 0 (lowest Brightness], 0xFF is highest Brightness */
+	// ret += nt35510_write_reg(dev, NT35510_CMD_WRCABCMB, &nt35510_reg34[1], 1);
+	ret += nt35510_write_reg_val(dev, NT35510_CMD_WRCABCMB, 0xff);
+
+	/* Send Command Display On */
+	ret += nt35510_write_reg(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+
+	/* Send Command GRAM memory write (no parameters)
+	   this initiates frame write via other DSI commands sent by
+	   DSI host from LTDC incoming pixels in video mode */
+	ret += nt35510_write_reg(dev, NT35510_CMD_RAMWR, NULL, 0);
+
+	return ret;
+}
+
+static int nt35510_write(const struct device *dev, uint16_t x, uint16_t y,
+			 const struct display_buffer_descriptor *desc, const void *buf)
+{
+	return -ENOTSUP;
+}
+
+static int nt35510_blanking_on(const struct device *dev)
+{
+	const struct nt35510_config *cfg = dev->config;
+	int ret;
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 0);
+		if (ret) {
+			LOG_ERR("Disable backlight failed! (%d)", ret);
+			return ret;
+		}
+	}
+	return nt35510_write_reg(dev, MIPI_DCS_SET_DISPLAY_OFF, NULL, 0);
+}
+
+static int nt35510_blanking_off(const struct device *dev)
+{
+
+	const struct nt35510_config *cfg = dev->config;
+	int ret;
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 1);
+		if (ret) {
+			LOG_ERR("Enable backlight failed! (%d)", ret);
+			return ret;
+		}
+	}
+	return nt35510_write_reg(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+}
+
+static int nt35510_set_brightness(const struct device *dev, const uint8_t brightness)
+{
+	return nt35510_write_reg(dev, MIPI_DCS_SET_DISPLAY_BRIGHTNESS, &brightness, 1);
+	return 0;
+}
+
+static void nt35510_get_capabilities(const struct device *dev,
+				     struct display_capabilities *capabilities)
+{
+	const struct nt35510_config *cfg = dev->config;
+	struct nt35510_data *data = dev->data;
+
+	memset(capabilities, 0, sizeof(struct display_capabilities));
+	capabilities->x_resolution = cfg->width;
+	capabilities->y_resolution = cfg->height;
+	capabilities->supported_pixel_formats = data->dsi_pixel_format;
+	capabilities->current_pixel_format = data->dsi_pixel_format;
+	capabilities->current_orientation = data->orientation;
+}
+
+static int nt35510_set_pixel_format(const struct device *dev,
+				    const enum display_pixel_format pixel_format)
+{
+	struct nt35510_data *data = dev->data;
+
+	data->dsi_pixel_format = pixel_format;
```

Comment:
Should there be a check here to verify the requested pixel format is actually supported?

Reply:
Yes you are right, and "enum display_pixel_format" must also be mapped into MIPI_DSI_PIXFMT_xxx values
Also realized that resulting color is hardcoded in mipi_dsi_stm32_attach (dsi_stm32.c):
"vcfg->ColorCoding = STM32_DSI_INIT_PIXEL_FORMAT" which comes from LTDC color configuration
But i will fix the verification and mapping to DSI colors in display_nt35510.c